### PR TITLE
Feature/info

### PR DESCRIPTION
**Pull request by AquaXV**

### CODE REVIEW

File: src/Commands/Modules/Moderation/EmoteCommand.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+using Discord;
+using Discord.Commands;
+
+namespace BrackeysBot.Commands
+{
+    public partial class ModerationModule : BrackeysBotModule
+    {
+        [Command("limitemote")]
+        [Summary("List the emote limits of the current channel")]
+        [Remarks("limitemote")]
```
Comment (YilianSource) [inline src/Commands/Modules/Moderation/EmoteCommand.cs]: We can omit the `Remarks` if `HideFromHelp` is present.

File: src/Core/Extensions/ChannelExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using Discord;
+
+namespace BrackeysBot
+{
+    public static class ChannelExtensions
+    {
+        public static string Linkable(this IMessageChannel channel)
```
Comment (YilianSource) [inline src/Core/Extensions/ChannelExtensions.cs]: Doesn't `IMessageChannel` implement `IMentionable`?
`channel.Mention` should suffice.

File: src/Core/Extensions/EmoteExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+using Discord;
+
+namespace BrackeysBot
+{
+    public static class EmoteExtensions
```
Comment (YilianSource) [inline src/Core/Extensions/EmoteExtensions.cs]: Out of curiosity, what are these extensions for?
Is there nothing built-in?

File: src/Services/ReactionLimiterService.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        public async Task CheckReactionAsync(Cacheable<IUserMessage, ulong> cacheable, ISocketMessageChannel channel, SocketReaction reaction) 
+        {
+            Dictionary<ulong, List<string>> restricted = _dataService.Configuration.EmoteRestrictions;
+
+            if (restricted != null) 
+            {
+                List<string> limitedToEmotes = restricted.GetValueOrDefault(channel.Id);
+
+                if (limitedToEmotes.Count > 0 && !limitedToEmotes.Contains(reaction.Emote.StringVal())) 
+                    await reaction.Message.Value.RemoveReactionAsync(reaction.Emote, reaction.User.Value);
```
Comment (YilianSource) [inline src/Services/ReactionLimiterService.cs]: Does this always work? Are there complications with rate limitations?

File: src/Commands/Modules/Moderation/EmoteCommand.cs (hunk 1 above)
Comment (AquaXV) [reply src/Commands/Modules/Moderation/EmoteCommand.cs]: Removed.

File: src/Core/Extensions/ChannelExtensions.cs (hunk 2 above)
Comment (AquaXV) [reply src/Core/Extensions/ChannelExtensions.cs]: 'IMessageChannel' does not contain a definition for 'Mention' and the best extension method overload 'UserExtensions.Mention(ulong)' requires a receiver of type 'ulong' [BrackeysBot]

So apparently not.

File: src/Core/Extensions/EmoteExtensions.cs (hunk 3 above)
Comment (AquaXV) [reply src/Core/Extensions/EmoteExtensions.cs]: The API has 3 kinds of emotes; 
Regular emotes the user can add like :Kappa:
Unicode emojis available by default like :rage: 
Animated emotes the user can add like :ablobrage:
Regular and Animated are handled as Emote : IEmote where unicode emoji are handled as Emoji : IEmote. IEmote only provides GetName() method. As such I had to add this extension to get them easily parseable.

File: src/Core/Extensions/ChannelExtensions.cs (hunk 2 above)
Comment (YilianSource) [reply src/Core/Extensions/ChannelExtensions.cs]: Ah right, that was `ITextChannel`.
Is there a reason to use `IMessageChannel` over `ITextChannel`?

File: src/Services/ReactionLimiterService.cs (hunk 4 above)
Comment (AquaXV) [reply src/Services/ReactionLimiterService.cs]: I don't see us using this in the announcement channel. The target channel has no rate limitation issues I'd imagine.

Worst case we get rate limited and it is deleted belated.

File: src/Core/Extensions/ChannelExtensions.cs (hunk 2 above)
Comment (AquaXV) [reply src/Core/Extensions/ChannelExtensions.cs]: Is ITextChannel supported as parameter input of commands?

Comment (YilianSource) [reply src/Core/Extensions/ChannelExtensions.cs]: Any implementation of `IChannel` should work.

Comment (AquaXV) [reply src/Core/Extensions/ChannelExtensions.cs]: Done.

**Pull request merged**